### PR TITLE
lib/streamaggr: pick sample with bigger timestamp or value on deduplicator

### LINKED ISSUE #5643: stream aggregation: deduplication should keep the sample with biggest timestamp on interval
### Is your feature request related to a problem? Please describe

Currently, `--remoteWrite.streamAggr.dedupInterval` option in vmagent reuses [last](https://docs.victoriametrics.com/stream-aggregation.html#last) aggregation output for deduplication. But this differs from how VictoriaMetrics deduplicates samples and may result into incorrect output.

### Describe the solution you'd like

It would be better not to keep the last received value on deduplication interval, but keep the sample with the biggest timestamp on the interval or the biggest value on timestamp conflicts - see https://docs.victoriametrics.com/#deduplication

This should prevent from improper deduplication results when there are more than metrics source producing duplicates. For example, if there is an HA pair of clients pushing data to vmagent then there is a chance that the pushed values will differ. This is extremely important for counters calculation, because it may result in counter to behave more like gauge if clientA will send value lower than clientB.

### PR DESCRIPTION
**Pull request by AndrewChubatiuk**
fixes #5643

### CODE REVIEW

**Comment by hagen1778** [comment]
Related PR https://github.com/VictoriaMetrics/VictoriaMetrics/pull/5931

File: lib/streamaggr/dedup_test.go [hunk 1]
```diff
@@ -23,8 +23,8 @@ func TestDedupAggrSerial(t *testing.T) {
 		da.pushSamples(samples)
 	}
 
-	if n := da.sizeBytes(); n > 3_400_000 {
-		t.Fatalf("too big dedupAggr state before flush: %d bytes; it shouldn't exceed 3_400_000 bytes", n)
+	if n := da.sizeBytes(); n > 4_200_000 {
```
[the remark this answers is not in the document]
Comment (AndrewChubatiuk) [reply lib/streamaggr/dedup_test.go:26]: tests are failing due to added timestamp to a structure

File: lib/streamaggr/streamaggr.go [hunk 2]
```diff
@@ -775,9 +777,13 @@ func (a *aggregator) Push(tss []prompbmarshal.TimeSeries, matchIdxs []byte) {
 				// Skip NaN values
 				continue
 			}
+			if sample.Timestamp == 0 {
```
Comment (hagen1778) [inline lib/streamaggr/streamaggr.go:780]: Under what conditions could this happen? From the quick check, all ingestion protocols already set timestamp by default.

**Review by hagen1778 — CHANGES_REQUESTED** [review]
Looks good! But please see my comment above.
Please also add line to changelog https://docs.victoriametrics.com/CHANGELOG.html
and update https://docs.victoriametrics.com/stream-aggregation/#deduplication since description contradicts with this change.

File: lib/streamaggr/streamaggr.go (hunk 2 above)
Comment (AndrewChubatiuk) [reply lib/streamaggr/streamaggr.go:780]: relied on this commit description
https://github.com/VictoriaMetrics/VictoriaMetrics/commit/7acc54025e8ca596147082da29af4a7245026959

Comment (hagen1778) [reply lib/streamaggr/streamaggr.go:780]: I see. I checked protocols here https://github.com/VictoriaMetrics/VictoriaMetrics/tree/master/lib/protoparser and most of them set timestamp if it is empty. Datadog protocol implementation doesn't do this, but I assume non-zero timestamp is set by datadog agent instead.
I suggest we skip this by now, and add it back once we need it.

**Review by hagen1778 — APPROVED** [review]
LGTM!

**Comment by valyala** [comment]
@AndrewChubatiuk , see also follow-up commits at b3d84489ec4451448d2759a6cd0569228f1fbf7d and cbd80efcc144200b265ae16c1aa1ba559319636f .

**Pull request merged**